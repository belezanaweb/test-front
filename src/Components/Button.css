.btn {
  font-family: "Helvetica Neue", "Helvetica", "Arial", "sans-serif";
  font-weight: 600;
  font-size: calc(16px + 1vmin);

  text-transform: uppercase;

  background: #ff6a00;
  color: #ffffff;

  width: 100%;
  height: 60px;

  border: 0;
  border-radius: 4px;

  box-shadow: #d45a00 0 3px 0, 1px 5px 3px rgba(35, 26, 26, 0.25);

  margin-top: 10px;
  margin-bottom: 10px;
}

.btn:hover {
  background: rgb(212, 90, 0);
  box-shadow: unset;
  border: unset;
}
