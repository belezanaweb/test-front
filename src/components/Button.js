import React from 'react'; 
import styled from 'styled-components'; 

const Buttons = styled.button`
    height: 60px;
    width: 341px;
    border-radius: 3px;
    background-color: #FF6C00;
    box-shadow: inset 0 -3px 0 0 #D45A00, 0 2px 4px 0 rgba(0,0,0,0.25);
    margin-top: 3.5vh;
`
const Name = styled.div`
    height: 24px;
    /* width: 339px; */
    color: #FFF;
    font-family: "Helvetica Neue";
    font-size: 20px;
    font-weight: 700;
    letter-spacing: 0.5px;
    line-height: 24px;
    text-align: center;
`

const Button = (props) => {
    return (
        <Buttons>
            <Name>
                {props.children}
            </Name>
        </Buttons>
    )
}

export default Button;