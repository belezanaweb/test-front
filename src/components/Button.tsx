import React from 'react'
import { useHistory } from 'react-router-dom'

import { styled, keyframes } from '@/style/stitches.config'

const onHover = keyframes({
  '0%': { backgroundColor: '$orange100' },
  '100%': { backgroundColor: '$orange300' }
})

const ButtonStyled = styled('button', {
  backgroundColor: '$orange100',
  width: '-webkit-fill-available',
  color: 'white',
  fontSize: 20,
  fontWeight: 'bold',
  textTransform: 'uppercase',
  marginX: '10px',
  marginY: '5px',
  paddingY: '15px',
  paddingX: '15px',
  border: 'none',
  borderBottom: '4px solid $orange300',
  borderRadius: '3px',
  '&:hover': {
    animation: `${onHover} 300ms`,
    backgroundColor: '$orange300'
  }
})

interface OwnProps {
  text: string
  href?: string
  isSubmit?: boolean
}

const Box = (props: OwnProps): React.ReactElement => {
  const { text, href, isSubmit = false } = props

  const history = useHistory()

  const onClick = (): void => {
    if (href) {
      history.push(href)
    }
  }

  if (isSubmit) {
    return (
      <div>
        <ButtonStyled type="submit" data-testid="button">
          {text}
        </ButtonStyled>
      </div>
    )
  }

  return (
    <ButtonStyled onClick={href ? onClick : undefined} data-testid="button">
      {text}
    </ButtonStyled>
  )
}

export default Box
