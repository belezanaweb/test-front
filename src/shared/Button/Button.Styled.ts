import styled, { css, DefaultTheme } from 'styled-components';

const variantStyles = (theme: DefaultTheme, variant = 'primary') => ({
  primary: css`
    background: ${theme.colors.primary};
    color: #fff;
  `,
  secondary: css`
    background: ${theme.colors.secondary};
    color: #fff;
  `,
}[variant]);

export const ButtonStyled = styled.button<{ variant: string }>`
  font-family: 'Arial', sans-serif;
  border-radius: 4px;
  padding: 16px 8px;
  border: 0;
  display: block;
  width: 100%;
  font-size: 18px;
  font-weight: 800;
  cursor: pointer;
  ${({ theme, variant }) => variantStyles(theme, variant)}
`;