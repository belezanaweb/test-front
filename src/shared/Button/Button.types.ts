export type ButtonsProps = {
  onClick?: React.ButtonHTMLAttributes<unknown>['onClick']
  variant?: string
  form?: string
}