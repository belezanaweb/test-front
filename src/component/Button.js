import React from 'react'
import * as S from './Button.styles'

const Button = ({ text, onClick, ...props }) => {
  return (
    <S.StyledButton onClick={onClick} {...props}>
      {text}
    </S.StyledButton>
  )
}

export default Button
