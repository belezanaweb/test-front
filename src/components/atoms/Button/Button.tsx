import * as Styles from './Button.styles'
interface ButtonProps {
  text: string;
  type: ('button' | 'submit');
  onClick: () => void;
  disabled?: boolean;
}

const Button: React.FC<ButtonProps> = ({ text, disabled = false, type, onClick }) => {
  return (
    <Styles.Container>
      <Styles.Button type={type} disabled={disabled} onClick={onClick}>
        {text}
      </Styles.Button>
    </Styles.Container>

  )
}

export default Button
