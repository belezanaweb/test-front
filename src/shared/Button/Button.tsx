import { PropsWithChildren } from 'react'

import { ButtonStyled } from './Button.Styled'
import { ButtonsProps } from './Button.types'

export const Button = ({ onClick, form, variant = 'primary', children }: PropsWithChildren<ButtonsProps>) => {
  return (
    <ButtonStyled onClick={onClick} form={form} variant={variant}>
      {children}
    </ButtonStyled>
  )
}
