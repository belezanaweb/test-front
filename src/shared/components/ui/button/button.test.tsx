import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import Button from './index';

describe('<Button />', () => {
  it('should render component with <button> tag', () => {
    render(<Button>button</Button>, {});
    expect(screen.getByRole('button'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<Button>button</Button>, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
