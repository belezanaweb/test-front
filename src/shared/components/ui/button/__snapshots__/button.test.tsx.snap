// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render component with default props and match snapshot 1`] = `
.c0 {
  border-radius: 4px;
  border: none;
  cursor: pointer;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  padding: 1rem 0.8rem;
  background: #ff6c00;
  box-shadow: inset 0 -3px 0 0 #d45a00,0 2px 4px 0 #00000040;
  color: #fff;
  text-transform: uppercase;
  font-size: 2rem;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

.c0:hover {
  box-shadow: inset 0 -50px 0 0 #d45a00,0 2px 4px 0 #00000040;
}

<button
  class="c0"
  type="button"
>
  button
</button>
`;
